### ISSUE TITLE
RemoveCombineCallback does not function

### ISSUE THREAD

**Issue by crater0** [jkulawik/tdd-issues#21]
Note: This has always been broken and as such is a low priority issue.

`RemoveCombineCallback` does not actually remove the callback, this seems to be because the loop never actually runs.

~~It also seems like triggering an inventory combination action was intended to remove the callback once triggered, this does not correctly function for the same reason.~~
Disregard, I had the autoremove variable set wrong in my test map, this behavior works as expected when the AutoRemove callback variable is set to true.

**Comment by jkulawik** [comment]
In other words, you mean that mods might expect trigerred inventory combinations to not be removed?

If yes, I think the base game is also likely to expect this.

**Comment by jkulawik** [comment]
Also I understood that the issue was that the loop checks for iterator begin instead of end. Which to me looks like the loop doesn't run at all.

![image](https://github.com/jkulawik/tdd-issues/assets/56608092/149dc63d-a100-4e38-a27e-b3e3ada7777a)

**Comment by crater0** [comment]
You are right, it is because the loop never triggers. I am blind and misunderstood what was causing the error initially.

**Comment by crater0** [comment]
> In other words, you mean that mods might expect trigerred inventory combinations to not be removed?
> 
> If yes, I think the base game is also likely to expect this.

Made an error, the autoremove behavior does actually function correctly. Just `RemoveCombineCallback` is bugged.

**Comment by jkulawik** [comment]
Should be fixed in 14695213
